frontends/web/src/app.tsx

```diff
@@ -14,50 +14,63 @@
  * limitations under the License.
  */
 
-// @ts-nocheck
-
-import { i18nEditorActive } from './i18n/i18n';
-import TranslationHelper from './components/translationhelper/translationhelper';
 import { Component, h } from 'preact';
 import { getCurrentUrl, route } from 'preact-router';
-import { apiGet } from './utils/request';
-import { apiWebsocket } from './utils/websocket';
-import { Update } from './components/update/update';
+import { Alert } from './components/alert/Alert';
+import { Confirm } from './components/confirm/Confirm';
+import { Container } from './components/container/container';
 import Sidebar from './components/sidebar/sidebar';
-import Container from './components/container/Container';
-import { DeviceSwitch } from './routes/device/deviceswitch';
+import TranslationHelper from './components/translationhelper/translationhelper';
+import { Update } from './components/update/update';
+import { i18nEditorActive } from './i18n/i18n';
 import Account from './routes/account/account';
-import Send from './routes/account/send/send';
-import Receive from './routes/account/receive/receive';
+import { AddAccount } from './routes/account/add/addaccount';
 import Info from './routes/account/info/info';
-import Settings from './routes/settings/settings';
-import ElectrumSettings from './routes/settings/electrum';
+import Receive from './routes/account/receive/receive';
+import Send from './routes/account/send/send';
+import { Devices, DeviceSwitch } from './routes/device/deviceswitch';
 import ManageBackups from './routes/device/manage-backups/manage-backups';
-import { Alert } from './components/alert/Alert';
-import { Confirm } from './components/confirm/Confirm';
-import { AddAccount } from './routes/account/add/addaccount';
+import ElectrumSettings from './routes/settings/electrum';
+import Settings from './routes/settings/settings';
+import { apiGet } from './utils/request';
+import { apiWebsocket } from './utils/websocket';
+
+interface AccountInterface {
+    coinCode: string;
+    code: string;
+    name: string;
+    blockExplorerTxPrefix: string;
+}
 
-export class App extends Component {
-    state = {
-        accounts: null,
+interface State {
+    accounts: AccountInterface[];
+    accountsInitialized: boolean;
+    deviceIDs: string[];
+    devices: Devices;
+    activeSidebar: boolean;
+}
+
+export class App extends Component<{}, State> {
+    public state = {
+        accounts: [],
         accountsInitialized: false,
         deviceIDs: [],
         devices: {},
         activeSidebar: false,
-    }
+    };
+
+    private unsubscribe?: () => void = undefined;
```

Comment:
if we know that `unsubscribe` will definitely be defined (in this case `componentDidMount`?), you can declare it as `private unsubscribe!`. Although this works as is so if typescript yells at you for this I would leave it 👍